### PR TITLE
Support custom key material when creating KMS keys

### LINKED ISSUE #8910: enhancement request: Support preseeding the new KMS provider
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Enhancement description

It would be great if the new KMS provider could be pre-seeded with actual key material.

The former `local-kms` provider support pre-seeding, but also lacked several endpoints. While new provider recently [added support for custom key IDs](https://github.com/localstack/localstack/commit/82bc739e2286c2a29bfcdf9c4e6ad3856adb9694), there are scenarios where this is not sufficient.

In particular, when developing against web3 smart contracts, having static key material for testing can be necessary to establish trust between local services and the blockchain.

Unfortunately, using `local-kms` isn't an option due to the lack of HMAC support.

### 🧑‍💻 Implementation

Ideally the key material could be loaded from a local file mounted into the localstack container, similar to how `local-kms` worked. Another option could be an API that can be called via an initialization hook.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jalaziz**
## Motivation

Add support for creating KMS keys with custom key material using custom tags. This is similar to an existing feature that allows the KMS Key ID to be set when creating KMS keys. This is necessary when working with external systems and decryption or HMACs must work across localstack reboots.

Fixes #8910

## Changes

* Introduces a new special tag `_custom_key_material_` that can be used when creating a KMS key to set the key material of the key.
* Refactors the existing `_custom_id_` so that it handled in the `KmsKey` model.

## Testing

I've introduced two new unit tests to ensure the custom key material is working for symmetric and HMAC KMS keys.

### CODE REVIEW

**Comment by jalaziz** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jalaziz** [comment]
recheck

**Comment by jalaziz** [comment]
recheck

File: tests/aws/services/kms/test_kms.py
Comment (sannya-singal) [inline tests/aws/services/kms/test_kms.py:143]: I would suggest to use some encryption method to generate these bytes string rather than using a static value, it primarily gives a broader context to the test being added.

File: tests/aws/services/kms/test_kms.py
Comment (sannya-singal) [inline tests/aws/services/kms/test_kms.py:115]: Can we give add some more context here, like rather than using a static value we can actually use some encoding method to generate these bytes which can be used by the below section of code.

**Review by sannya-singal — CHANGES_REQUESTED** [review]

**Comment by sannya-singal** [comment]
Hi @jalaziz, thanks for contributing to localstack 🚀. These are amazing set of changes 🙂👏

I have added few comments, rest LGTM!

File: tests/aws/services/kms/test_kms.py
Comment (jalaziz) [reply tests/aws/services/kms/test_kms.py:143]: Thanks for the feedback. I purposefully didn't want to use the kms encrypt calls because it wouldn't really test the static encryption key (the test could pass if the custom key was ignored).

That being said, I could generate the encrypted text directly instead to provide more context.

Comment (sannya-singal) [reply tests/aws/services/kms/test_kms.py:143]: I am not sure what you mean by generating encrypted text directly, can you explain this a bit more?

Comment (jalaziz) [reply tests/aws/services/kms/test_kms.py:143]: Sorry, what I mean is calling the cryptography library directly in the test.

The purpose of this change and the tests is to ensure that secrets encrypted with the custom secret can be decrypted by the same secret across localstack restarts.

To ensure this, ideally the encrypted cipher text is static or generated in a way that is very stable.

Comment (sannya-singal) [reply tests/aws/services/kms/test_kms.py:143]: Do you mean using something like:

`from cryptography.hazmat.primitives import hashes` 

to encrypt?

Comment (jalaziz) [reply tests/aws/services/kms/test_kms.py]: Yup exactly. I've just pushed a commit with the test changes.

**Comment by sannya-singal** [comment]
@jalaziz thanks for fixing reviews! LGTM! I would request you to rebase your branch and then I can merge it! 🎉

**Comment by jalaziz** [comment]
@sannya-singal done!

**Comment by sannya-singal** [comment]
Thanks for fixing this @jalaziz 👏 🚀. Can you kindly request a re-review:
image
as this is blocking the merge. Thanks!

**Review by sannya-singal — APPROVED** [review]

**Pull request merged**
